R/dependencies.R

```diff
@@ -852,6 +858,114 @@ update_with_direct_deps <- function(project = ".",
   )
 }
 
+#' Recursively discover and build, check and install
+#' internal dependencies
+#'
+#' It discovers all dependencies starting from the repositories,
+#' determines the installation order and then
+#' builds, checks and installs them in order.
+#'
+#' @md
+#' @inheritParams rec_checkout_internal_deps
+#' @param steps (`character` vector) subset of "build", "check", "install";
+#'   useful to skip checking for example
+#' @param rcmd_args (`list`) with names `build`, `check`,
+#'   `install` which are vectors that are passed as separate arguments
+#'   to the `R CMD` commands
+#' @param artifact_dir (`character`) directory where build
+#'   tarball and logs go to
+#' @export
+#' @examples
+#' \dontrun{
+#' build_check_install_repos(
+#'   list(list(repo = "insightsengineering/teal", host = "https://github.com")),
+#'   feature = "main",
+#'   direction = "upstream",
+#'   local_repos = data.frame(repo = "insightsengineering/teal",
+#'   host = "https://github.com", directory = "../scratch1/teal/",
+#'   stringsAsFactors = FALSE)
+#' )
+#' build_check_install_repos(
+#'   list(list(repo = "maximilian_oliver.mordig/stageddeps.electricity",
+#'   host = "https://code.roche.com")),
+#'   feature = "master",
+#'   direction = "upstream",
+#'   local_repos = data.frame(repo = "maximilian_oliver.mordig/stageddeps.electricity",
+#'   host = "https://code.roche.com",
+#'   directory = "../example_ecosystem/stageddeps.electricity/",
+#'   stringsAsFactors = FALSE),
+#'   artifact_dir = "/tmp/test112"
+#' )
+#'
+#' # to install all packages
+#' build_check_install_repos(someArgs, steps = "install")
+#' # alternatively with slightly different arguments (e.g. dry_run),
+#' # also adds commit SHA
+#' install_deps(someArgs, direction = c("upstream", "downstream"))
+#' }
+build_check_install_repos <- function(repos_to_process, feature,
+                                      direction = c("upstream", "downstream"),
+                                      local_repos = get_local_pkgs_from_config(),
+                                      verbose = 0,
+                                      steps = c("build", "check", "install"),
+                                      rcmd_args = list(check = c("--no-manual")),
+                                      artifact_dir = tempfile()) {
```

Comment:
Do we want the default to be a tempfile()? When you run install or build within R Studio it puts the tar file in the parent direction of the package and at the sample place a folder  called <<package_name>>.00check (or something like that) to store the output of the check